src/detect-engine-siggroup.c

```diff
@@ -92,9 +92,12 @@ static SigGroupHeadInitData *SigGroupHeadInitDataAlloc(uint32_t size)
         return NULL;
 
     /* initialize the signature bitarray */
-    sghid->sig_size = size;
-    if ((sghid->sig_array = SCCalloc(1, sghid->sig_size)) == NULL)
+    size = sghid->sig_size = size + 16 - (size % 16);
+    void *ptr = SCMallocAligned(sghid->sig_size, 16);
+    if (ptr == NULL)
         goto error;
+    memset(ptr, 0, size);
+    sghid->sig_array = ptr;
```

Comment:
Sorry for being late.
Q: Why don't we only align the bitarray if SSE2 instruction set is available? `sig_size` would be different for example in such a case and we'll malloc lesser bytes.
e.g. for a size 10,
`master`: sig_size = 10; sizeof(sig_array) = 10 * sizeof(u8)
after this PR: sig_size = 16; sizeof(sig_array) = 16 * sizeof(u8) even when SSE2 is not available

Reply:
ok. that works then